src/compiler/core.ts

```diff
@@ -967,6 +983,20 @@ namespace ts {
                         subpattern += directorySeparator;
                     }
 
+                    if (usage !== "exclude") {
+                        // The * and ? wildcards should not match directories or files that start with . if they
+                        // appear first in a component. Dotted directories and files can be included explicitly
+                        // like so: **/.*/.*
+                        if (component.indexOf("*") === 0) {

```

Comment:
Can you use `component.charCodeAt(0) === CharacterCodes.asterisk` here? It would be inefficient to look at every character of a component when we only care about the first one.